nucleus/payara-modules/healthcheck-core/src/main/java/fish/payara/nucleus/healthcheck/admin/SetHealthCheckConfiguration.java

```diff
@@ -0,0 +1,207 @@
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import javax.inject.Inject;
+import javax.validation.constraints.Min;
+
+import org.glassfish.api.ActionReport;
+import org.glassfish.api.I18n;
+import org.glassfish.api.Param;
+import org.glassfish.api.admin.AdminCommand;
+import org.glassfish.api.admin.AdminCommandContext;
+import org.glassfish.api.admin.CommandLock;
+import org.glassfish.api.admin.CommandRunner;
+import org.glassfish.api.admin.ExecuteOn;
+import org.glassfish.api.admin.ParameterMap;
+import org.glassfish.api.admin.RestEndpoint;
+import org.glassfish.api.admin.RestEndpoints;
+import org.glassfish.api.admin.RuntimeType;
+import org.glassfish.api.admin.ServerEnvironment;
+import org.glassfish.config.support.CommandTarget;
+import org.glassfish.config.support.TargetType;
+import org.glassfish.hk2.api.PerLookup;
+import org.glassfish.hk2.api.ServiceLocator;
+import org.glassfish.internal.api.Target;
+import org.jvnet.hk2.annotations.Service;
+import org.jvnet.hk2.config.ConfigSupport;
+import org.jvnet.hk2.config.TransactionFailure;
+
+import com.sun.enterprise.config.serverbeans.Config;
+
+import fish.payara.nucleus.healthcheck.HealthCheckService;
+import fish.payara.nucleus.healthcheck.configuration.HealthCheckServiceConfiguration;
+import fish.payara.nucleus.notification.TimeUtil;
+import fish.payara.nucleus.notification.configuration.NotifierType;
+import fish.payara.nucleus.notification.log.LogNotifierConfiguration;
+
+/**
+ * Service to configure the {@link HealthCheckServiceConfiguration} of the {@link Target}.
+ *
+ * Updating the service configuration also updates the {@link LogNotifierConfiguration} accordingly.
+ *
+ * @author jan
+ * @since 5.191
+ */
+@Service(name = "set-healthcheck-configuration")
+@PerLookup
+@CommandLock(CommandLock.LockType.NONE)
+@I18n("healthcheck.configure")
+@ExecuteOn({RuntimeType.DAS,RuntimeType.INSTANCE})
+@TargetType({CommandTarget.DAS, CommandTarget.STANDALONE_INSTANCE, CommandTarget.CLUSTER, CommandTarget.CLUSTERED_INSTANCE, CommandTarget.CONFIG})
+@RestEndpoints({
+    @RestEndpoint(configBean = HealthCheckServiceConfiguration.class,
+            opType = RestEndpoint.OpType.POST,
+            path = "set-healthcheck-configuration",
+            description = "Enables/Disables Health Check Service")
+})
+public class SetHealthCheckConfiguration implements AdminCommand {
+
+    @Inject
+    ServerEnvironment server;
+
+    @Inject
+    private Logger logger;
+
+    @Inject
+    HealthCheckService healthCheck;
+
+    @Inject
+    ServiceLocator serviceLocator;
+
+    @Inject
+    private Target targetUtil;
+    private Config targetConfig;
+
+    @Param(name = "dynamic", optional = true, defaultValue = "false")
+    private boolean dynamic;
+
+    @Param(name = "target", optional = true, defaultValue = "server-config")
+    private String target;
+
+    @Param(name = "enabled")
+    private boolean enabled;
+
+    @Param(name = "trace-enabled", optional = true)
```

Comment:
I was writing the documentation and noticed I wasn't happy with the name either. That said I also don't like the old name much better. If I get this feature right it allows to keep previous runs of the health check service. The entries can be limited by two factors: The number (size) and the age (timeout). I think both these are really confusingly named. From the user's perspective I think of this as `--keep-results`, `max-kept-results`, `max-result-age`. Do you think this would be an improvement or do you rather want to go back to old names with `historical`. In that case should they even use camel case instead of dash?

Reply:
I think I'd still leave it with the old name for now.
It is something possibly worth coming back to though, as your argument of defining the number of results to keep and for how long does have some merit over defining the size of the trace store.

Reply:
Done.